plugins/targetlocker/inmemory/inmemory.go

```diff
@@ -132,70 +166,72 @@ func broker(lockRequests, unlockRequests, checkLocksRequests <-chan request, don
+// InMemory locks targets in an in-memory map.
 type InMemory struct {
-	lockRequests, unlockRequests, checkLocksRequests chan request
+	lockRequests, unlockRequests, checkLocksRequests chan *request
 	done                                             chan struct{}
-	timeout                                          time.Duration
+	// timeout set on each lock request
+	timeout time.Duration
 }
 
-func newReq(targets []*target.Target) request {
+func newReq(jobID types.JobID, targets []*target.Target) request {
 	return request{
 		targets: targets,
+		owner:   jobID,
 		err:     make(chan error),
 	}
 }
 
-// Lock locks the specified targets by doing nothing.
+// Lock locks the specified targets.
 func (tl *InMemory) Lock(jobID types.JobID, targets []*target.Target) error {
 	log.Infof("Trying to lock %d targets", len(targets))
-	req := newReq(targets)
+	req := newReq(jobID, targets)
 	req.timeout = tl.timeout
-	tl.lockRequests <- req
+	tl.lockRequests <- &req
 	return <-req.err
 }
 
-// Unlock unlocks the specified targets by doing nothing.
+// Unlock unlocks the specified targets.
 func (tl *InMemory) Unlock(jobID types.JobID, targets []*target.Target) error {
 	log.Infof("Trying to unlock %d targets", len(targets))
-	req := newReq(targets)
-	tl.unlockRequests <- req
+	req := newReq(jobID, targets)
+	tl.unlockRequests <- &req
 	return <-req.err
 }
 
 // RefreshLocks extends the lock duration by the internally configured timeout. If
 // the owner is different, the request is rejected.
 func (tl *InMemory) RefreshLocks(jobID types.JobID, targets []*target.Target) error {
 	log.Infof("Trying to refresh locks on %d targets by %s", len(targets), tl.timeout)
-	req := newReq(targets)
+	req := newReq(jobID, targets)
 	req.timeout = tl.timeout
 	// refreshing a lock is just a lock operation with the same owner and a new
 	// duration.
-	tl.lockRequests <- req
+	tl.lockRequests <- &req
 	return <-req.err
 }
 
-// CheckLocks tells whether all the targets are locked. They all are, always. It
-// also returns the ones that are not locked.
+// CheckLocks tells whether all the targets are locked, and returns a slice of
+// locked and a slice of unlocked targets.
 func (tl *InMemory) CheckLocks(jobID types.JobID, targets []*target.Target) (bool, []*target.Target, []*target.Target) {
 	log.Infof("Checking if %d target(s) are locked by job ID %d", len(targets), jobID)
-	req := newReq(targets)
-	tl.checkLocksRequests <- req
+	req := newReq(jobID, targets)
+	tl.checkLocksRequests <- &req
 	if err := <-req.err; err != nil {
 		// TODO the target.Locker.CheckLocks interface has to change to return an
 		// error as well, because lock checking can fail, e.g. when using
 		// external services.
 		// Just log an error for now.
 		log.Warningf("Error when trying to unlock targets: %v", err)
 	}
-	return len(req.notLocked) != len(targets), req.locked, req.notLocked
+	return len(req.locked) == len(targets), req.locked, req.notLocked
```

Comment:
this check was wrong - the returned boolean means "all the targets you specified are locked by you". Adjusted to do just that